SPECIFICATION.md

```diff
@@ -348,7 +348,7 @@ interested in realistic examples of FonotBene fonts, please take a look at the
 ```fontobene
 [format]
 format = FontoBene
-format_version = 1.0.0
```

Comment:
What's the difference between 1.0.0 and 1.0?

Reply:
Consistency with other examples ;)

Reply:
OTOH we follow Semantic Versioning, and SemVer generally uses 3 section versions, doesn't it?

Reply:
Yes it does, but it's too late to change it because LibrePCB does a string comparison with "1.0" :grin:

Reply:
If that's the case it's not standards compliant, because 1.0 and 1.0.0 is equivalent in SemVer, and we write that FontoBene follows SemVer without specifying how the version string looks like :slightly_smiling_face: 

I'd suggest to add support for both 1.0 and 1.0.0 in LibrePCB and then to explicitly specify that version numbers need 3 segments.

Reply:
Yes we should do that. I'll still merge this PR now as from the Fontobene point of view it doesn't matter if the version is specified as "1.0" or "1.0.0" as both are equivalent (we can change it later as it doesn't break the specs)...

Reply:
It depends. If we explicitly state that 3 segments are required, then that would make old files that don't follow this recommendation invalid, making it a breaking change that would require v2 to be released :wink: 

If you want to be correct, I'd probably change that in this PR.

Reply:
> if we explicitly state that 3 segments are required,

we shouldn't do that ;)

> making it a breaking change that would require v2 to be released

Yes, but this PR is about to release v1, so I musn't introduce any breaking change.

> If you want to be correct, I'd probably change that in this PR.

But then the official examples of Fontobene can't be read by LibrePCB. I would like to first fix it in LibrePCB and adjust the Fontobene examples later.

Reply:
Opened issue here: https://github.com/fontobene/fontobene-qt5/issues/1

Reply:
> Yes, but this PR is about to release v1, so I musn't introduce any breaking change.

?? :smile: The point of doing changes *before* 1.0 is that it's still possible.

I don't particularly care in this case, but what's the use of asking for comments if the comments are then not discussed until we reached a consensus? :wink: 

I would have preferred requiring 3 segments, because then a version equivalence check is a simple string comparison. Now a parser is required. But I guess when determining compatibility a parser is required anyways, so it doesn't really matter.

Reply:
Actually I just checked the SemVer spec, and contrary to what I thought I remembered 1.0 is not actually a valid SemVer version. So we now have an internally conflicting spec :wink: 

I'd suggest simply fixing that directly without a version increment, that should be totally fine since nobody except LibrePCB uses it so far.

Reply:
Sorry, I don't have the nerves to discuss about such a small detail ;)

> The point of doing changes before 1.0 is that it's still possible.

No, v1.0 *was* somehow already released with LibrePCB a few weeks ago, we just forgot to update this repository accordingly, so I did it now.

> I would have preferred requiring 3 segments

Me too.

> Now a parser is required.

It is anyway, as you already recognized.

Btw, I'm open to approve a PR which adjusts the specs accordingly :wink:

Reply:
Well, you did the change that changed the spec from 1.0.0 to 1.0, it would have been the easiest to simply revert that change before merging... But well, I'll do a PR.

Reply:
Ha, you're right, sorry! I thought you changed 1.0.0-WIP to 1.0, that's the reason why I was a bit irritated that you insisted not to change it back :smile:

Reply:
> No, v1.0 was somehow already released with LibrePCB a few weeks ago, we just forgot to update this repository accordingly, so I did it now.

Since when is LibrePCB released? Last I saw was rc-2